lib/model/folder.go

```diff
@@ -64,6 +70,83 @@ func newFolder(model *Model, cfg config.FolderConfiguration) folder {
 	}
 }
 
+func (f *folder) Serve() {
+	l.Debugln(f, "starting")
+	defer l.Debugln(f, "exiting")
+
+	defer func() {
+		f.scan.timer.Stop()
+		f.setState(FolderIdle)
+	}()
+
+	pause := f.basePause()
+	pullFailTimer := time.NewTimer(0)
+	<-pullFailTimer.C
+
+	if f.FSWatcherEnabled && f.CheckHealth() == nil {
+		f.startWatch()
+	}
+
+	initialCompleted := f.initialScanFinished
+
+	for {
+		select {
+		case <-f.ctx.Done():
+			return
+
+		case <-f.pullScheduled:
+			pullFailTimer.Stop()
+			select {
+			case <-pullFailTimer.C:
+			default:
+			}
+
+			if !f.puller.pull() {
+				// Pulling failed, try again later.
+				pullFailTimer.Reset(pause)
+			}
+
+		case <-pullFailTimer.C:
+			if !f.puller.pull() {
+				// Pulling failed, try again later.
+				pullFailTimer.Reset(pause)
+				// Back off from retrying to pull with an upper limit.
+				if pause < 60*f.basePause() {
+					pause *= 2
```

Comment:
Out of this pr context, but the fact we never reset this is worrying.

Reply:
lol